element_calcium_imaging/imaging.py

```diff
@@ -1544,6 +1544,101 @@ def make(self, key):
             raise NotImplementedError("Unknown/unimplemented method: {}".format(method))
 
 
+@schema
+class PostProcessingQualityMetrics(dj.Computed):
+    definition = """
+    -> Segmentation
+    """
+
+    class MaskMetrics(dj.Part):
+        definition = """
+        -> master
+        -> Segmentation.Mask
+        ---
+        mask_area: float  # Mask area in square micromete. Stringer & Pachitariu (2019).
+        roundness: float  # Roundness between 0 and 1, closer to 1 the rounder. Tegtmeier et al. (2018).
+        """
+
+    class FluorescenceTraceMetrics(dj.Part):
+        definition = """
+        -> master
+        -> Fluorescence.Trace
+        ---
+        skewness: float   # Stringer & Pachitariu (2019)
+        variance: float   # Stringer & Pachitariu (2019)
+        """
+
+    def make(self, key):
+        from scipy.stats import skew
+
+        (
+            mask_xpixs,
+            mask_ypixs,
+            mask_weights,
+            fluorescence,
+            fluo_channels,
+            mask_ids,
+            mask_npix,
+            px_height,
+            px_width,
+            um_height,
+            um_width,
+        ) = (Segmentation.Mask * scan.ScanInfo.Field * Fluorescence.Trace & key).fetch(
+            "mask_xpix",
+            "mask_ypix",
+            "mask_weights",
+            "fluorescence",
+            "fluo_channel",
+            "mask",
+            "mask_npix",
+            "px_height",
+            "px_width",
+            "um_height",
+            "um_width",
+        )
+
+        self.insert1(key)
+
+        roundnesses = np.empty_like(mask_xpixs)
+        for i, (mask_xpix, mask_ypix, mask_weight) in enumerate(
+            zip(mask_xpixs, mask_ypixs, mask_weights)
+        ):
+            eigen_values = np.linalg.eigvals(
+                np.cov(mask_xpix, mask_ypix, aweights=mask_weight)
+            )
+            roundnesses[i] = eigen_values.mean() / eigen_values.max()
```

Comment:
```suggestion
       norm_mean = lambda x: x.mean() / x.max()
       roundness = np.array([
           norm_mean(np.linalg.eigvals(np.cov(x, y, aweights=w))) 
           for x, y, w in zip(mask_xpixs, mask_ypixs, mask_weights)])       
```

Reply:
The variable name should be plural, `roundnesses`.  Otherwise it will break the insert line further down the code.

Reply:
The `insert` statement below is not yet correct anyway.

Reply:
changed to `roundnesses` although we could come up with a more natural name.